src/main/java/metrics_influxdb/api/protocols/HttpInfluxdbProtocol.java

```diff
@@ -0,0 +1,63 @@
+package metrics_influxdb.api.protocols;
+
+public class HttpInfluxdbProtocol implements InfluxdbProtocol {
+    public final static String DEFAULT_HOST = "127.0.0.1";
+    public final static int DEFAULT_PORT = 8086;
+    public final static String DEFAULT_DATABASE = "metrics";
+    private final String user;
+    private final String password;
+    private final String host; 
+    private final int port;
+    private final boolean secured;
+    private final String database;
+    
+    public HttpInfluxdbProtocol(String host, int port, String user, String password, String db) {
+        super();
+        this.host = host;
+        this.port = port;
+        this.user = user;
+        this.password = password;
+        this.database = db;
+        this.secured = (user != null) && (password != null);
+    }
+    
+    public HttpInfluxdbProtocol(String host) {
+        this(host, DEFAULT_PORT);
+    }
+    
+    public HttpInfluxdbProtocol(String host, int port) {
+        this(host, port, null, null);
+    }
+    
+    public HttpInfluxdbProtocol(String host, int port, String database) {
+        this(host, port, null, null, database);
+    }
+    
+    public HttpInfluxdbProtocol() {
+        this(DEFAULT_HOST, DEFAULT_PORT);
+    }
+    
+    public HttpInfluxdbProtocol(String host, int port, String user, String password) {
+        this(host, port, user, password, DEFAULT_DATABASE);
+    }
+    
+    public String getUser() {
```

Comment:
It's more a matter of taste of how your API should look like : either you expose public final fields or you expose getters which allows you to refactor your internals, to do calculations, to introduce decoration, interception, ... But is is probably not the place for such a debate.